### PR TITLE
Add Fedora24, OpenSUSE42, and Ubuntu 16.10

### PR DESCRIPTION
**Pull request by eerhardt**
test ci please

### CODE REVIEW

**Comment by eerhardt** [comment]
@piotrpMSFT @livarcocc @jgoshi @jonsequitur @krwq - please take a look.  I'd like to merge this today so I can start standing up the VSO build defs and the Jenkins legs

**Comment by eerhardt** [comment]
test ci please

File: test/dotnet-new.Tests/GivenThatIWantANewAppWithSpecifiedType.cs [hunk 1]
```diff
@@ -30,6 +30,13 @@ public class GivenThatIWantANewAppWithSpecifiedType : TestBase
             string projectType,
             bool useNuGetConfigForAspNet)
         {
+            if (language == "F#" && !EnvironmentInfo.HasSharedFramework("netcoreapp1.0"))
+            {
+                // F# requires netcoreapp1.0 to be present in order to build
+                // TODO: eerhardt log a bug
```
Comment (eerhardt) [inline test/dotnet-new.Tests/GivenThatIWantANewAppWithSpecifiedType.cs]: I'll fix this to point to https://github.com/dotnet/netcorecli-fsc/issues/76

File: TestAssets/TestProjects/AppWithDirectAndToolDep/AppWithDirectAndToolDep.csproj [hunk 2]
```diff
@@ -17,7 +18,4 @@
       <Version>1.0.0</Version>
     </DotNetCliToolReference>
   </ItemGroup>
-  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
-    <DefineConstants>$(DefineConstants);RELEASE</DefineConstants>
-  </PropertyGroup>
 </Project>
```
Comment (piotrpMSFT) [inline TestAssets/TestProjects/AppWithDirectAndToolDep/AppWithDirectAndToolDep.csproj]: nit: newline

File: TestAssets/TestProjects/AppWithToolDependency/AppWithToolDependency.csproj [hunk 3]
```diff
@@ -1,23 +1,16 @@
...
       <Version>1.0.0</Version>
     </DotNetCliToolReference>
     <DotNetCliToolReference Include="dotnet-PreferCliRuntime">
       <Version>1.0.0</Version>
     </DotNetCliToolReference>
   </ItemGroup>
-  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
-    <DefineConstants>$(DefineConstants);RELEASE</DefineConstants>
-  </PropertyGroup>
 </Project>
```
Comment (piotrpMSFT) [inline TestAssets/TestProjects/AppWithToolDependency/AppWithToolDependency.csproj]: same here

File: test/EndToEnd/GivenDotNetUsesMSBuild.cs [hunk 4]
```diff
@@ -19,7 +19,7 @@ public void ItCanNewRestoreBuildRunCleanMSBuildProject()
             {
                 string projectDirectory = directory.Path;
 
-                string newArgs = "console --debug:ephemeral-hive";
+                string newArgs = "console -f netcoreapp1.1 --debug:ephemeral-hive";
```
Comment (piotrpMSFT) [inline test/EndToEnd/GivenDotNetUsesMSBuild.cs]: Should the default here become 1.1? @mlorbetske does the latest template update make that change?

Comment (mlorbetske) [inline test/EndToEnd/GivenDotNetUsesMSBuild.cs]: Yep, this is the new default, but given that it was quite confusing trying to track down some of the issues that we saw in getting our PR to go green, I think it's best to be explicit here (as it's much harder to search for the absence of something to fix up the next time the default changes)

**Pull request merged**